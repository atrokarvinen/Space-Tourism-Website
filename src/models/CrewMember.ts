import { Image } from "./Image";

export interface CrewMember {
  name: string;
  images: Image;
  role: string;
  bio: string;
}
