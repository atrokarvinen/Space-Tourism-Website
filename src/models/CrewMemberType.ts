import { Image } from "./Image";

export interface CrewMemberType {
  name: string;
  images: Image;
  role: string;
  bio: string;
}
