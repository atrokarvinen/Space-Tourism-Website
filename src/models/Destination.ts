import { Image } from "./Image";

export interface Destination {
  name: string;
  images: Image;
  description: string;
  distance: string;
  travel: string;
}
