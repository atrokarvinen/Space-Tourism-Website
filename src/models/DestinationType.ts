import { Image } from "./Image";

export interface DestinationType {
  name: string;
  images: Image;
  description: string;
  distance: string;
  travel: string;
}
