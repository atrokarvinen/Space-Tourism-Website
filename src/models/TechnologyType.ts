export interface TechnologyType {
  name: string;
  images: { portrait: string; landscape: string };
  description: string;
}
