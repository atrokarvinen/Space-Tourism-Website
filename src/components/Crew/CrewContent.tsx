import style from "./Crew.module.scss";

import React, { ReactElement } from "react";

import { CrewMemberType } from "../../models/CrewMemberType";

interface CrewContentProps {
  crewMember: CrewMemberType;
}

export default function CrewContent({
  crewMember,
}: CrewContentProps): ReactElement {
  const { name, role, bio } = crewMember;
  return (
    <>
      <h4 className={style.role}>{role.toUpperCase()}</h4>
      <h3 className={style.name}>{name.toUpperCase()}</h3>
      <p>{bio}</p>
    </>
  );
}
