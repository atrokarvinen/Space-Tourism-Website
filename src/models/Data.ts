import { CrewMember } from "./CrewMember";
import { Destination } from "./Destination";
import { Technology } from "./Technology";

export interface Data {
    destinations: Destination[];
    crew: CrewMember[];
    technology: Technology[];
}