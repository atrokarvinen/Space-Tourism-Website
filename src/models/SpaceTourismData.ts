import { CrewMemberType } from "./CrewMemberType";
import { DestinationType } from "./DestinationType";
import { TechnologyType } from "./TechnologyType";

export interface SpaceTourismData {
  destination: DestinationType[];
  crew: CrewMemberType[];
  technology: TechnologyType[];
}
