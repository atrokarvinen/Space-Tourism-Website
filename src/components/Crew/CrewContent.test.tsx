import { render } from "@testing-library/react";
import { CrewMemberType } from "../../models/CrewMemberType";
import CrewContent from "./CrewContent";

it("matches snapshot", () => {
  const crewMember: CrewMemberType = {
    name: "john doe",
    images: { png: "", webp: "" },
    bio: "fancy biography",
    role: "captain",
  };
  const { container } = render(<CrewContent crewMember={crewMember} />);

  expect(container.innerHTML).toMatchSnapshot();
});
