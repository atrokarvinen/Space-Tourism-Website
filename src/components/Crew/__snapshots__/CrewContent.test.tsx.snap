// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `"<h4 class=\\"role\\">CAPTAIN</h4><h3 class=\\"name\\">JOHN DOE</h3><p>fancy biography</p>"`;
